domains/certificates/Query.API/API/ContractService/ContractServiceImpl.cs

```diff
@@ -115,41 +115,52 @@ private static MeteringPointType Map(MeterType type)
         return null;
     }
 
-    public async Task<SetEndDateResult> SetEndDate(Guid id, UserDescriptor user, DateTimeOffset? newEndDate, CancellationToken cancellationToken)
+    public async Task<SetEndDateResult> SetEndDate(List<(Guid id, UnixTimestamp? newEndDate)> contracts, UserDescriptor user, CancellationToken cancellationToken)
     {
         var meteringPointOwner = user.Subject.ToString();
-        var contract = await unitOfWork.CertificateIssuingContractRepo.GetById(id, cancellationToken);
-
-        if (contract == null)
-        {
-            return new NonExistingContract();
-        }
 
-        if (contract.MeteringPointOwner != meteringPointOwner)
+        foreach (var (id, newEndDate) in contracts)
         {
-            return new MeteringPointOwnerNoMatch();
-        }
-
-        if (newEndDate.HasValue && newEndDate <= contract.StartDate)
-        {
-            return new EndDateBeforeStartDate(contract.StartDate, newEndDate.Value);
+            var contract = await unitOfWork.CertificateIssuingContractRepo.GetById(id, cancellationToken);
```

Comment:
We should not GetById in a for loop. We should move this up and fetch all Certificates in 1 database call.